### DISCUSSION TITLE
Flow - Resume on external trigger

### DISCUSSION THREAD

**Discussion by ernst86913** [directus/directus#16797] [Feature Requests]
Hi - is there any operation available where I can make a flow wait for an external trigger and after the trigger resume the flow?

thanks,
Roman

**Comment by raphourbe** [comment]
Hi,

To my knowledge, you can probably do this with the **[Run script](https://docs.directus.io/configuration/flows/operations.html#run-script)** if you know what you are looking for and can code it.

For example, I use it to create Stripe customers this way (it can be an `async` function):

```js
module.exports = async function(data) {
    const stripe = require('stripe')('sk_API');
    
    const customer = await stripe.customers.create({
        email: data.read_user.email,
        name: data.read_user_id.enterprise,
        address: {
            city: data.read_user_id.address_city,
            country: data.read_user_id.address_country,
            line1: data.read_user_id.address_line,
            postal_code: data.read_user_id.address_postal_code
        },
        preferred_locales: [
            "fr-FR"
        ],
        metadata: {
        	directusId: data.read_user_id.user_id
        }
    });
    return customer;
}
```

Otherwise, if you do not need any data from your external trigger, and you know how long for this external operation to end, you might use the **[Sleep](https://docs.directus.io/configuration/flows/operations.html#sleep)** operation.

**Reply by ernst86913** [reply]
Hi, I am referring to real external triggers on which the flow should be waiting for... examples:
- Internal Events (e.g. another object es created / updated / deleted....)... Example: The workflow creates a task that needs manual intervention by someone... the person picks up the task in the UI and completes it... and then the workflow should resume
- External Events (e.g. the flow needs to wait for an incoming email... or the flow needs to wait for an API call... good example for this would be customer onboarding / registration where the flow needs to wait for the customer mail verification...
KR
Roman

**Reply by raphourbe** [reply]
For your Internal Events, if a person has to do something in the UI, you can add in Collections or Items a button to trigger a Flow (see [directus doc](https://docs.directus.io/configuration/flows/triggers.html#manual)). Which means you can maybe create two Flows. First one which does the first part of the job. Then a second one that wait for a manual trigger.

For the External Events, I have implemented on my app an [activation email](https://rphl.dev/blog/building-saas-with-directus-and-stripe-part3). The way I do it though is different from what you mention. I send thanks to a Flow an activation email that send the customer to my front, where I have a call to Directus to update a value in my DB (account_activated), that allows the customer to enter my app. By the way, when I update the account_activated boolean, it triggers a new Flow.

However, if you absolutely need to have a Flow to activate the account for example, you can create a flow triggered thanks to a link. I explain how to do it [there](https://rphl.dev/blog/sending-params-to-directus-flow), and here is the [directus documentation part](https://docs.directus.io/configuration/flows/triggers.html#webhook). So your activation mail could be the link to this flow with a parameter that is an encrypted id of your customer, or maybe a token that you cancel at the end of the flow (see the post about the activation email for the token solution).

Finally, you can use the [Another Flow trigger](https://docs.directus.io/configuration/flows/triggers.html#another-flow) mentioned in the documentation to chain Flows, and pass data to one another.

**Comment by ernst86913** [comment]
Thanks a lot for all your comments and insights - really helpful.

so - only real way to do the external one is by splitting one workflow into two… not nice - but a workaround.

Still -  many major workflow tools (eg Camunda) provide a “wait for trigger” step which resumes the flow… anyway - workaround for now should definitely work for me. Thanks

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
